WordPress/gutenberg
# Add widgets customize inspector

## Description

Close #29286

The inspector panel is done by manually inserting a _fake_ section into the DOM via `createPortal`. The transition effect is achieved by manually triggering classes in `useEffect`. The implementation is pretty hacky, I'm not proud of it..., but it works? At least for now. We should also handle the i18n and a bunch of other things though, but I want to get a first round of review, in case I missed something really obvious 😅.

## TODOs

- [ ] Pressing <kbd>esc</kbd> should close the inspector.
- [ ] Scrolling in the inspector should trigger the header to be sticky.
- [ ] Opening the inspector should move focus to its first focusable element (the back button).
- [x] Closing the inspector should move the focus back to the block.

## How has this been tested?



1. Enable "Widgets in customizer"
2. Go to Appearance -> Customize -> Widgets -> select any widget area
3. Select any block
4. Click the "more" settings in the customizer, and select "Show more settings"
5. The inspector panel should slide in

## Screenshots 

https://user-images.githubusercontent.com/7753001/110887144-babde300-8324-11eb-8449-c341fd4f97c3.mp4

## Types of changes




New feature

## Checklist:
